test/test_unix_socket/receiver.py

```diff
@@ -23,23 +23,18 @@ def run_dummy_listener(socket_path=DEFAULT_UNIX_SOCKET_PATH):
         client_socket.connect(socket_path)
         print("Connected to server.")
 
+        buffer = ""
         while True:
```

Comment:
This looks CPU intensive. Does recv block until data is available or does it just return nothing?

Reply:
See above. tldr: yes, it blocks